### PR TITLE
Allow TokenAuthentication subclasses

### PR DESCRIPTION
**Pull request by jpic**
Stop trigger of the check for TokenAuthentication if you have a TokenAuthentication subclass.

PS: lovely app, great for thrashing away AWS Cognito

### CODE REVIEW

File: rest_registration/checks.py [hunk 1]
```diff
@@ -96,7 +96,10 @@ def verification_from_check():
 def token_auth_config_check():
     return implies(
         registration_settings.LOGIN_RETRIEVE_TOKEN,
-        TokenAuthentication in api_settings.DEFAULT_AUTHENTICATION_CLASSES,
+        any(filter(
```
Comment (apragacz) [inline rest_registration/checks.py]: What about:
```python
        any(
            issubclass(cls, TokenAuthentication)
            for cls in api_settings.DEFAULT_AUTHENTICATION_CLASSES
        )
```
It is more readable IMO and it has the same semantic meaning.

**Comment by apragacz** [comment]
@jpic If you want I can merge your changes at current state and refactor it later. What do you think?

**Comment by jpic** [comment]
Thanks for your review, pushed the requested change, have fun !

**Comment by apragacz** [comment]
@jpic I think there may be an unecessary comma next to `issubclass(cls, TokenAuthentication)`. Could you remove it? I think this should fix the problem with failing linters and tests and I could merge the change right away!

**Comment by apragacz** [comment]
Removed the comma and merged. Thank you!

**Pull request merged**